### PR TITLE
MINIO_UPDATE takes no effect on gateway mode

### PR DESCRIPTION
**Pull request by TestMsr**
## Description
control the update check on gateway mode

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-zoned.sh | :x: [more...](#4548b77c-81f3-488a-ad4a-f1318ba5fc82) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#27b430aa-d4df-4109-a13e-5f4bcd0f0877) |
<a name="27b430aa-d4df-4109-a13e-5f4bcd0f0877"></a>
### 12262-74d26e3/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:30982
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ed8d28bea23f:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 31 seconds
{
  "name": "awscli",
  "duration": 2558,
  "function": "aws --endpoint-url http://minio-dev8.minio.io:30982 s3api copy-object --bucket awscli-mint-test-bucket-23478 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-23478/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 46 seconds
(7/15) Running minio-dotnet tests ... done in 43 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 13 seconds
{
  "args": {},
  "duration": 214,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 2 minutes and 3 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 142,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-34gujch, objectName = minio-java-test-43b5so-copy, resource = /minio-java-test-34gujch/minio-java-test-43b5so-copy, requestId = 167D0FA5F2247DCC, hostId = 72c677ce-9a3a-4e7e-a1bb-5c5525128d68)\nrequest={method=PUT, url=http://minio-dev8.minio.io:30982/minio-java-test-34gujch/minio-java-test-43b5so-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-qrdjum/minio-java-test-43b5so\nHost: minio-dev8.minio.io:30982\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210508T101046Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210508/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 416\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Sat, 08 May 2021 10:10:46 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 167D0FA5F2247DCC\nX-Xss-Protection: 1; mode=block\nDate: Sat, 08 May 2021 10:10:46 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 47 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 42 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2650",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-13769/s3cmd-test-object-21651 bcb27fc0-ad3a-4f00-845c-c4a15605229a\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id bcb27fc0-ad3a-4f00-845c-c4a15605229a put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
<a name="4548b77c-81f3-488a-ad4a-f1318ba5fc82"></a>
### 12262-74d26e3/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31895
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f5a9a1b970b6:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 1 minutes and 25 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 32 seconds
(8/15) Running minio-dotnet tests ... done in 48 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 12 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 57 seconds
(11/15) Running minio-js tests ... done in 55 seconds
(12/15) Running minio-py tests ... FAILED in 2 minutes and 34 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-b8da5bd4-40bd-4038-95a3-f10cbe8b22ff",
    "object_name": "9866630d-7e62-423d-a9ee-a3ff94455a3f"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-b8da5bd4-40bd-4038-95a3-f10cbe8b22ff, request_id: 167D0FE00317BC6F, host_id: 69c71d61-e24f-456c-80ee-4e5afd8d36fd, bucket_name: minio-py-test-b8da5bd4-40bd-4038-95a3-f10cbe8b22ff",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-b8da5bd4-40bd-4038-95a3-f10cbe8b22ff, request_id: 167D0FE00317BC6F, host_id: 69c71d61-e24f-456c-80ee-4e5afd8d36fd, bucket_name: minio-py-test-b8da5bd4-40bd-4038-95a3-f10cbe8b22ff\n",
  "duration": 6574
}
(12/15) Running s3cmd tests ... done in 20 seconds
(13/15) Running s3select tests ... done in 5 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/gateway-main.go [hunk 1]
```diff
@@ -360,8 +360,11 @@ func StartGateway(ctx *cli.Context, gw Gateway) {
 	// Prints the formatted startup message once object layer is initialized.
 	if !globalCLIContext.Quiet {
```
Comment (harshavardhana) [inline cmd/gateway-main.go:361]:
```suggestion
	if !globalCLIContext.Quiet && !globalInplaceUpdateDisabled {
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/gateway-main.go (hunk 1 above)
Comment (TestMsr) [reply cmd/gateway-main.go:361]: it seems that the poroduction ready and startup message be quiet while inplcace update diabled?

**Comment by harshavardhana** [comment]
@TestMsr thanks for the PR - I created another PR that supersedes this change and simplifies a few things that needs to be done https://github.com/minio/minio/pull/12493 

So closing this PR as it might not be needed anymore.

**Pull request closed without merging**